### PR TITLE
Updated Bio.phenomics

### PR DESCRIPTION
**Pull request by peterjc**
How does this look @mgalardini to replace #830?

I've created ``test_phenomics_scipy.py`` for the fitting test which needs SciPy, although there is a warning here you might be able to help with:

```
$ python2.7 test_phenotype_scipy.py
test_WellRecord (__main__.TestPhenoMicro)
Test basic functionalities of WellRecord objects. ... /home/xxx/lib/python2.7/site-packages/scipy/optimize/minpack.py:690: OptimizeWarning: Covariance of the parameters could not be estimated
  category=OptimizeWarning)
ok

----------------------------------------------------------------------
Ran 1 test in 0.774s

OK
```

```
$ python3.5 test_phenotype_scipy.py
test_WellRecord (__main__.TestPhenoMicro)
Test basic functionalities of WellRecord objects. ... /home/xxx/lib/python3.5/site-packages/scipy/optimize/minpack.py:690: OptimizeWarning: Covariance of the parameters could not be estimated
  category=OptimizeWarning)
ok

----------------------------------------------------------------------
Ran 1 test in 0.842s

OK
```

These are both using SciPy 0.17.1 in case that helps.

### CODE REVIEW

**Comment by peterjc** [comment]
Assuming you're happy with the changes. I'll probably tweak this a little to squash some of my most recent commits into a cleaner history before merging this to the master.

**Comment by peterjc** [comment]
I'd missed the import fix and splitting of the test had already been done on #830.

The API changes here might still be worth considering...

**Comment by mgalardini** [comment]
Thanks for (better) implementing some of the things that were in the other PR. I'm happy with all the proposed changes, except maybe raising an exception if no function can be fitted. PM data is quite noisy, and ~10% of the wells will have no function fitted. I think that a warning is less 'traumatic' for the user.

So, I'll implement your changes (minus the exception if you are ok with that) in the other PR, possibly tonight.
Marco

File: Doc/Tutorial/chapter_phenotype.tex [hunk 1]
```diff
@@ -0,0 +1,271 @@
+\chapter{Bio.phenotype: analyse phenotypic data}
+\label{chapter:phenotype}
+
+This chapter gives an overwiev of the functionalities of the
```
Comment (cbrueffer) [inline Doc/Tutorial/chapter_phenotype.tex]: Typo "overwiev".

File: Doc/Tutorial/chapter_phenotype.tex [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+high-throughput phenotypic experiments produced by the
+\href{https://en.wikipedia.org/wiki/Phenotype_microarray}{Phenotype Microarray technology},
+but future developments may include other platforms and formats.
+
+\section{Phenotype Microarrays}
+\label{sec:phenotypemicroarrays}
+
+The \href{https://en.wikipedia.org/wiki/Phenotype_microarray}{Phenotype Microarray}
+is a technology that measures the metabolism of bacterial
+and eukaryotic cells on rhoughly 2000 chemicals, divided in twenty 96-well
```
Comment (cbrueffer) [inline Doc/Tutorial/chapter_phenotype.tex]: Typo, "rhoughly"

File: Doc/Tutorial/chapter_phenotype.tex [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+...
+PM01 96
+PM01 96
+PM09 96
+PM09 96
+\end{verbatim}
+
+The parser returns a series of PlateRecord objects, each one containing a series of WellRecord objects
+(holding each well's experimental data) arranged in 8 rows and 12 columns; each row is indicated by
+a uppercase charachter from A to H, while columns are indicated by a two digit number, from 01 to 12.
```
Comment (cbrueffer) [inline Doc/Tutorial/chapter_phenotype.tex]: Typo, "charachter"

File: Doc/Tutorial/chapter_phenotype.tex [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+ 132.95999999999998,
+ 136.95999999999998,
+ 140.0,
+ 142.0,
+ nan]
+\end{verbatim}
+
+\subsubsection{Control well subtraction}
+Many Phenotype Microarray plates contain a control well (usually A01), that is a well where the media shouldn't support
+any growth; the low signal produced by this well can be subtracted to the other wells.
```
Comment (cbrueffer) [inline Doc/Tutorial/chapter_phenotype.tex]: "to the other wells" -> "from the other wells"

**Comment by mgalardini** [comment]
Thanks @cbrueffer, I'll add those corrections as well

**Pull request closed without merging**